src/BlocklyInterface/robotblocks/general/GeneralBlocks.js

```diff
@@ -0,0 +1,377 @@
+/*
+ * Code taken and lightly modified from FTC Blockly application. Modifications
+ * cover global variable usage from other files that are not suitable to be
+ * directly copied across, or are themselves transpiled from unknown sources
+ */
+
+// The below lines are the additions to the FTC original general block
+// definition to cover transpiled code parts
+import Blockly from "blockly";
+const miscIdentifierForJavaScript = "miscAccess";
+const functionColor = 289;
+const commentColor = 200;
+
+function createNonEditableField(label) {
+  var field = new Blockly.FieldTextInput(label);
+  field.CURSOR = "";
+  field.showEditor_ = function (opt_quietInput) {};
+  return field;
+}
+
+// FTC copied code begins here
+
+Blockly.Blocks["comment"] = {
+  init: function () {
+    this.appendDummyInput().appendField(
+      new Blockly.FieldTextInput(""),
+      "COMMENT"
+    );
+    this.setPreviousStatement(true);
+    this.setNextStatement(true);
+    this.setColour(commentColor);
+  },
+};
+
+Blockly.JavaScript["comment"] = function (block) {
+  return "";
+};
+
+Blockly.JavaScript["comment"] = function (block) {
+  return "// " + block.getFieldValue("COMMENT") + "\n";
+};
+
```

Comment:
Why are there two implementation of this and many other blocks?

Reply:
Doh! That's my bad, I left in a few Java implementations re-routed to JS, when they should all be JS. Changing now